bdk-ffi/src/descriptor.rs

```diff
@@ -36,9 +35,9 @@ impl Descriptor {
         keychain_kind: KeychainKind,
         network: Network,
     ) -> Self {
-        let derivable_key = secret_key.inner_mutex.lock().unwrap();
+        let derivable_key = &secret_key.inner;
 
-        match derivable_key.deref() {
+        match derivable_key {
             BdkDescriptorSecretKey::XPrv(descriptor_x_key) => {
```

Comment:
@notmandatory I forgot to ask but does this seem like the best approach to you?

I'm now borrowing the `secret_key.inner`, and the `derivable_key` doesn't need to be dereferenced anymore. I just want to make sure I'm not messing up something with the ownership rules.

Reply:
Your approach looks good to me, since you only need to read the value borrowing it is the right thing to do.